pkg/sharing/sharing.go

```diff
@@ -184,6 +202,77 @@ func (s *Sharing) CreateRequest(inst *instance.Instance) error {
 	return couchdb.CreateNamedDocWithDB(inst, s)
 }
 
+// Revoke remove the credentials for all members, contact them, removes the
+// triggers and set the active flag to false.
+func (s *Sharing) Revoke(inst *instance.Instance) error {
+	var errm error
+
+	if !s.Owner {
+		return ErrInvalidSharing
+	}
+	for i := range s.Members {
+		if i != 0 {
+			if err := s.RevokeMember(inst, &s.Members[i], &s.Credentials[i-1]); err != nil {
+				multierror.Append(errm, err)
+			}
+		}
+	}
+	if s.WithPropagation() {
+		if err := s.RemoveTriggers(inst); err != nil {
+			return err
+		}
+	}
+	if err := RemoveReferences(inst, s.SID); err != nil {
+		return err
+	}
+	s.Active = false
+	if err := couchdb.UpdateDoc(inst, s); err != nil {
+		return err
+	}
+	return errm
+}
+
+// RemoveTriggers remove all the triggers associated to this sharing
+func (s *Sharing) RemoveTriggers(inst *instance.Instance) error {
+	var err error
+	sched := jobs.System()
+	// TODO: a view on WorkerType would be more efficient
```

Comment:
A third way would be to put the trigger IDs on the `Triggers` struct of `Sharing`. Currently, we have bool but it should be easy to modify it to have the IDs there.